### PR TITLE
GH-307: Automatically select n4js product to run when a JUnit Plugin test is started

### PR DESCRIPTION
**Pull request by qtran-n4**
# Task
Link: https://github.com/eclipse/n4js/issues/307

In order to test this do the following:
- Open Eclipse Installer. 
- Use the Oomph setup file: https://raw.githubusercontent.com/eclipse/n4js/GH-307-runconfig/releng/org.eclipse.n4js.targetplatform/N4JS.setup
- At the end of the Oomph setup process, you will eventually see the following log message in the console of the newly installed IDE:

```
0    [main] INFO  eConfigFileForSettingDefaultProduct  - File .../ws/.metadata/.plugins/org.eclipse.pde.core/.install_folders/1508744923642/configuration/config.ini has been created successfully!
From now on, the default product to run is set to org.eclipse.n4js.product.product
```

Try to execute any Plugin UI or SWTBot test and verify that `org.n4js.eclipse.product.product` is selected as product to run.

### CODE REVIEW

**Comment by qtran-n4** [comment]
@mmews-n4 Please review this PR. Thanks!

File: releng/org.eclipse.n4js.releng.utils/src/org/eclipse/n4js/runconfig/CreateConfigFileForSettingDefaultProduct.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.apache.log4j.Logger;
+
+/**
+ * This class access the variable ${target_home} and create a config.ini file specifying
+ * org.eclipse.n4js.product.product as the default product to run. This saves us from having to manually configure the
+ * product again and again whenever we trigger a test. GH-307. Attention: If you change this class, make sure to make a
```
Comment (mmews-n4) [inline releng/org.eclipse.n4js.releng.utils/src/org/eclipse/n4js/runconfig/CreateConfigFileForSettingDefaultProduct.java]: - omit the GH-307
- Add a \<p> before "Attention"
- consider to use <b>Attention:</b>
- ... make sure to _create_ ...

File: releng/org.eclipse.n4js.targetplatform/N4JS.setup [hunk 2]
```diff
@@ -517,10 +517,16 @@
...
+    <description>Launch that creates a config.ini</description>
+  </setupTask>
+  <stream name="GH-307-runconfig"
+      label="GH-307-runconfig"/>
   <logicalProjectContainer
       xsi:type="setup:ProjectCatalog"
       href="index:/org.eclipse.setup#//@projectCatalogs[name='com.github']"/>
   <description>N4JS provides an IDE for typed Javascript.</description>
-</setup:Project>
+</setup:Project>
```
Comment (mmews-n4) [inline releng/org.eclipse.n4js.targetplatform/N4JS.setup]: new line missing

File: releng/org.eclipse.n4js.releng.utils/src/org/eclipse/n4js/runconfig/CreateConfigFileForSettingDefaultProduct.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+		FileWriter writer = new FileWriter(file);
+		writer.write("eclipse.product=" + defaultProductId);
+		writer.close();
+		LOGGER.info("File" + filePath
+				+ " has been created successfully!\nFrom now on, the default product to run is set to " +
+				defaultProductId);
+	}
+
+}
```
Comment (mmews-n4) [inline releng/org.eclipse.n4js.releng.utils/src/org/eclipse/n4js/runconfig/CreateConfigFileForSettingDefaultProduct.java]: new line missing

File: releng/org.eclipse.n4js.releng.utils/src/org/eclipse/n4js/runconfig/CreateConfigFileForSettingDefaultProduct.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+			LOGGER.info("File " + filePath + " already exists.\nBy default, the product to run is set to " +
+					defaultProductId);
+			return;
+		}
+
+		FileWriter writer = new FileWriter(file);
+		writer.write("eclipse.product=" + defaultProductId);
+		writer.close();
+		LOGGER.info("File" + filePath
+				+ " has been created successfully!\nFrom now on, the default product to run is set to " +
```
Comment (mmews-n4) [inline releng/org.eclipse.n4js.releng.utils/src/org/eclipse/n4js/runconfig/CreateConfigFileForSettingDefaultProduct.java]: The "From now on" is always true. I would remove it. Also, the term "default product to run" is unclear.

**Comment by qtran-n4** [comment]
@mmews-n4 Thanks for your review! I have incorporated your review. I have also verified that it works fine with n4js-n4. Since build is green, I am merging now.

**Pull request merged**